src/main/java/eu/europeana/metis/sandbox/config/IndexingConfig.java

```diff
@@ -20,62 +20,124 @@
 
-  @Value("${sandbox.indexing.solr.zookeeper.default.collection:#{null}}")
-  private String zookeeperDefaultCollection;
+  @Value("${sandbox.preview.solr.zookeeper.default.collection:#{null}}")
+  private String zookeeperPreviewDefaultCollection;
 
-  @Value("${sandbox.indexing.solr.zookeeper.timeout:#{null}}")
-  private Integer zookeeperTimeoutInSecs;
+  @Value("${sandbox.preview.solr.zookeeper.timeout:#{null}}")
+  private Integer zookeeperPreviewTimeoutInSecs;
+
+  @Value("${sandbox.publish.mongo.hosts}")
+  private String[] mongoPublishHosts;
+
+  @Value("${sandbox.publish.mongo.ports}")
+  private int[] mongoPublishPorts;
+
+  @Value("${sandbox.publish.mongo.authentication-db:#{null}}")
+  private String mongoPublishAuthenticationDb;
+
+  @Value("${sandbox.publish.mongo.username:#{null}}")
+  private String mongoPublishUsername;
+
+  @Value("${sandbox.publish.mongo.password:#{null}}")
+  private String mongoPublishPassword;
+
+  @Value("${sandbox.publish.mongo.ssl-enable:#{null}}")
+  private Boolean mongoPublishEnableSSL;
+
+  @Value("${sandbox.publish.mongo.db}")
+  private String mongoPublishDb;
+
+  @Value("${sandbox.publish.solr.hosts}")
+  private String[] solrPublishHosts;
+
+  @Value("${sandbox.publish.solr.zookeeper.hosts:#{null}}")
+  private String[] zookeeperPublishHosts;
+
+  @Value("${sandbox.publish.solr.zookeeper.ports:#{null}}")
+  private int[] zookeeperPublishPorts;
+
+  @Value("${sandbox.publish.solr.zookeeper.chroot:#{null}}")
+  private String zookeeperPublishChroot;
+
+  @Value("${sandbox.publish.solr.zookeeper.default.collection:#{null}}")
+  private String zookeeperPublishDefaultCollection;
+
+  @Value("${sandbox.publish.solr.zookeeper.timeout:#{null}}")
+  private Integer zookeeperPublishTimeoutInSecs;
+
+  @Bean
+  Indexer previewIndexer()
+      throws URISyntaxException, SetupRelatedIndexingException, IndexerRelatedIndexingException {
+    return getIndexer(mongoPreviewHosts, mongoPreviewPorts, mongoPreviewDb,
+        mongoPreviewAuthenticationDb,
+        mongoPreviewUsername, mongoPreviewPassword, mongoPreviewEnableSSL, solrPreviewHosts,
+        zookeeperPreviewHosts, zookeeperPreviewPorts, zookeeperPreviewChroot,
+        zookeeperPreviewDefaultCollection, zookeeperPreviewTimeoutInSecs);
+  }
 
   @Bean
-  Indexer indexer()
+  Indexer publishIndexer()
       throws URISyntaxException, SetupRelatedIndexingException, IndexerRelatedIndexingException {
+    return getIndexer(mongoPublishHosts, mongoPublishPorts, mongoPublishDb,
+        mongoPublishAuthenticationDb,
+        mongoPublishUsername, mongoPublishPassword, mongoPublishEnableSSL, solrPublishHosts,
+        zookeeperPublishHosts, zookeeperPublishPorts, zookeeperPublishChroot,
+        zookeeperPublishDefaultCollection, zookeeperPublishTimeoutInSecs);
+  }
+
+  //Suppress: Methods should not have too many parameters warning
+  //We are okay with this method to ease configuration
+  @SuppressWarnings("squid:S107")
```

Comment:
Not critical, but we could perhaps bundle the mongo config and solr/zookeeper config?

Reply:
I am not sure to understand the suggestion. Could you expand? maybe an example

Reply:
Well I was thinking something in the line like:
getIndexer(getMongoConfiguration(mongoPublishHosts, mongoPublishPorts, mongoPublishDb,
        mongoPublishAuthenticationDb, mongoPublishUsername, mongoPublishPassword), getSolrConfiguration(solrPublishHosts,
        zookeeperPublishHosts, zookeeperPublishPorts, zookeeperPublishChroot,
        zookeeperPublishDefaultCollection, zookeeperPublishTimeoutInSecs))

And those two methods would return two classes that encapsulate the corresponding properties..

Was just an idea of how to fix that warning, but we don't have to follow that, as you prefer.

Reply:
I can do it, I will apply it in the ticket I am working now if that is ok
Thanks

Reply:
Yeah that's fine!